contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract DomainRegistry {
+    uint256 constant REQUIRED_DEPOSIT = 1 ether;
+
+    struct Domain {
+        uint256 deposit;
+        bool isRegistered;
+    }
+
+    mapping(string => Domain) public domains;
+
+    event DomainCreated(string tld, uint256 deposit);
+    event DomainReleased(string tld);
+
+    modifier hasRequiredDeposit(uint256 _requiredDeposit) {
```

Comment:
Any reason to have an argument in the modifier? Can it work without it?

Reply:
fxd